### DISCUSSION TITLE
How to rolling upgrade milvus from 2.2.X to 2.3.1 if we already use argo cd to maintain the helm repo?

### DISCUSSION THREAD

**Discussion by caozhuozi** [milvus-io/milvus#27685] [Q&A and General discussion]
Hi Milvus team,
From the doc, it is recommened that using an operator to do a rolling upgrade. As operator directy changes the Kubernetes resources, thoses changes, however, wont be synced  back to git repo.

So whats the best practice to do the rolling upgrade if we use argo cd？

Thanks

**Comment by LoveEachDay** [comment]
Milvus helm chart does not support graceful rolling upgrade because it cann't control the deployment restart order when upgrading.

**Reply by caozhuozi** [reply]
@LoveEachDay  Thanks for the quick reponse. Could you also tell me what is the proper order perform a graceful upgrade?

**Reply by caozhuozi** [reply]
@LoveEachDay 
According to the milvus operator code:
```go
func (r *MilvusReconciler) ReconcileAll(ctx context.Context, mc v1beta1.Milvus) error {
	reconcilers := []Func{
		r.ReconcileEtcd,
		r.ReconcileMsgStream,
		r.ReconcileMinio,
		r.ReconcileMilvus,
	}
	err := defaultGroupRunner.Run(reconcilers, ctx, mc)
	return errors.Wrap(err, "reconcile milvus")
}
```
I didnt see any evidence related to "order" here. I think all the 4 Components are reconcoled randomly in terms of order.

Besides, for the  specific `ReconcileMilvus` one, I also cant find any evidence:
```go
func (r *MilvusReconciler) ReconcileMilvus(ctx context.Context, mc v1beta1.Milvus) error {
	if !IsDependencyReady(mc.Status.Conditions) {
		return nil
	}

	if err := r.ReconcileConfigMaps(ctx, mc); err != nil {
		return fmt.Errorf("configmap: %w", err)
	}

	comReconcilers := []Func{
		r.ReconcilePVCs,
		r.ReconcileDeployments,
		r.ReconcileServices,
		r.ReconcileIngress,
		r.ReconcilePodMonitor,
	}
	err := defaultGroupRunner.Run(comReconcilers, ctx, mc)
	return errors.Wrap(err, "reconcile milvus")
}
```

Could you please clarify what is "order" means in the operator?

**Comment by LoveEachDay** [comment]
For milvus operator, you can create a `milvusUpgrade`cr.
